STRIPE change to use source instead of card

ptibogxiv opened the pull request:
use of more Gateways (3d secure, sepa direct debit…)  and add payment request API ( google pay, apple pay, Microsoft pay…)

ptibogxiv committed:
convert to use of source script instead of card

prepare use of more Gateways (3d secure, sepa direct debit…)  and payment request API ( google pay, apple pay, Microsoft pay…)
to do: recall of registered source of a thirdparty, save the source for a futur usage, more stripe gateway (with thirdparty country use)

ptibogxiv committed:
insert payment request API

add google pay, apple pay gateway

ptibogxiv committed:
fix

ptibogxiv committed:
RGPD compliance, save source only if case checked

ptibogxiv commented:
saving and pay function have some bugs, fix in progress

ptibogxiv committed:
Fix for saving source only if checkbox is checked

payment functionnal

ptibogxiv commented:
Bugs for saving and payment are fixed :-)

eldy committed:
Update newpayment.php

eldy commented on htdocs/public/payment/newpayment.php:
```diff
@@ -1542,97 +1548,3473 @@ class="stripe-button"
-	    <?php
-		print '</script>';
+<?php
+/* Copyright (C) 2001-2002	Rodolphe Quiedeville	<rodolphe@quiedeville.org>
```
Header of a php file in a middle of the php file ?

ptibogxiv replied:
@eldy fix of my bad copy/paste it's done

ptibogxiv committed:
fix bad paste/copy on file

ptibogxiv committed:
3D secure and apple/google pay functional

to do: optimise/create a return page url which retrieve if payment is ok or ko (informations it's not in URL)
to do: retrieve save sources and functionalize payment

ptibogxiv committed:
update for preventdefault use with owner-name

add support of owner's name (not required if apple/google pay)

ptibogxiv committed:
delete value of input

ptibogxiv committed:
choice between sources or new source

todo: more beautiful CSS integration

eldy committed:
Update newpayment.php

eldy committed:
Update newpayment.php

ptibogxiv commented:
@eldy form with source  and apple/google/Microsoft pay is functionnal. i Don't find the perfect internal css class in dolibarr for a better look/ergonomy.  why Don't rewrite new payment in a one screen form with all Gateway in the same list as on my form below
![capture](https://user-images.githubusercontent.com/3135833/39576378-3b845a9c-4ede-11e8-8ecb-3678afbd0544.PNG)

eldy commented:
There was so many change since this PR, that there was too much conflict that i failed to solve. Your help is welcome because being able to manage other stripe payment mode would be fantastic !

ptibogxiv commented:
@eldy  i will work to add full stripe source's support :

ptibogxiv committed:
WIP  more sources and functions

ptibogxiv committed:
Update newpayment.php

ptibogxiv committed:
ADD support of sources & PAYMENT REQUEST API

ptibogxiv commented:
@eldy i updated this commit to get support of more sources but I have pb with dolibarr CSS. 
For test google/ apple pay, you must add  a .well-know/ directory as Stripe's docs

![dolibarr](https://user-images.githubusercontent.com/3135833/47372195-8c0a2800-d6e9-11e8-9222-a6c2e10a0210.png)

eldy commented on htdocs/public/payment/newpayment.php:
```diff
@@ -432,16 +432,36 @@
+$src = \Stripe\Source::retrieve("$card",array("stripe_account" => $stripeacc));
+}
+else {
+$src = \Stripe\Source::create(array(
```
We do not provide the $metadata anymore when creating a card ?
